packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -266,24 +299,49 @@ class MouseTracker extends ChangeNotifier {
           )
         : <MouseTrackerAnnotation>{};
 
+    // Dispatch events (onEnter, onHover, onExit).
     _dispatchDeviceCallbacks(
       currentState: mouseState,
       nextAnnotations: nextAnnotations,
     );
 
+    // Change mouse cursor.
+    final MouseCursor cursor = _findDeviceCursor(nextAnnotations);
+    if (cursor != mouseState.cursor) {
+      _cursorManager.setDeviceCursor(device, cursor);
```

Comment:
If the requested cursor is not available we still pretend this was successful?

Reply:
If the cursor is not available, the action is considered successful, therefore older shells will not throw exceptions for not implementing a latest feature introduced in the framework. Only actual runtime errors are failures.